### DISCUSSION TITLE
bug with short mp3 duration with input.harbor

### DISCUSSION THREAD

**Discussion by masternet0** [savonet/liquidsoap#2901] [User Support]
Hello,

**Describe the bug**
there bug in input.harbor when send short duration less than 5 second
for example have 2 liquidsoap configurations 
liquidsoap_1.conf
liquidsoap_jingle.conf

liquidsoap_jingle.conf its run playlist every 1 minute to send it to input.harbor on this configuration liquidsoap_1.conf

liquidsoap_1.conf :
```
Jingle = input.harbor("jingle", port=8000, password="mypassword")
input = fallback(track_sensitive=false, [Jingle, autodj])
```

liquidsoap_jingle.conf :
```
jingle = playlist(loop=true, mode="normal", reload_mode="watch", "playlist.text")
jingle = delay(60.,jingle )
```

I see the log for both config works good and already the file send success but still the liquidsoap_1 play autodj not switch to jingle
and if the jingle duration more than 8 second it works

are the harbor feature check every more than 6 second or something ? so the check done after stream complete and close ?
this is last of log file for liquidsoap_1
```
2023/02/13 08:49:30 [input.harbor_3:3] New metadata chunk ? -- Hi Sweety I Love You SMS Ringtone, I Love You Notification Ringtone.
2023/02/13 08:49:32 [input.harbor_3:2] Feeding stopped: Ffmpeg_decoder.End_of_file.
2023/02/13 08:49:32 [threads:4] Thread "harbor source feeding" terminated (1 remaining).
```

so its start and completed and not switch to it
can you review this

Regards

**Version details**

- OS: ubuntu20
- Version : Liquidsoap 2.1.3

**Install method**
Did you install via `opam`? Using your distribution's packages? From source?
yes by opam

**Comment by toots** [comment]
Hi! Thanks for reporting.

`input.harbor` is not really suited for that kind of use I believe. It expects a long-term live source and, so, has an initial buffer to make sure it has enough data before becoming active.

Do you need the two liquidsoap instances? As I see it you should be able to do:
```
jingle = playlist(loop=true, mode="normal", reload_mode="watch", "playlist.text")
jingle = delay(60.,jingle )
input = fallback(track_sensitive=false, [Jingle, autodj])
```

**Comment by masternet0** [comment]
Hello,
Thanks for you response, but i did that because liquidsoap not support update data without kill the current process to activate changes
so if i used one liquidsoap instances if i updated on jingle path or added other type its should be restart the process and will effect to stream performance 

so i used 2 instances to update jingle instant any time without kill autodj process 
you have solution for this point ?
I hope soon add more flexibility for liquidsoap to update the config file without kill process and just reload the config to apply in next round auto
Regards

**Reply by toots** [reply]
The `playlist` operator can reload its content automatically when it changes. You should try to add `reload_mode="watch"` to the operator's arguments.

**Reply by toots** [reply]
Otherwise, if this does not work, I would suggest a queue of requests via `request.queue` to which you can push you jingles whenever you want the to be played.
